enhancements/sdk-generate-commands.md

```diff
@@ -0,0 +1,353 @@
+---
+title: README describing new command workflows for OLM enabled operators
+authors:
+  - "@varshaprasad96"
+  - "@estroz"
+reviewers:
+  - "@jlanford"
+  - "@dmesser"
+  - "@jmrodri"
+approvers:
+  - "@jlanford"
+  - "@dmesser"
+  - "@jmrodri"
+creation-date: 2020-04-09
+last-updated: 2020-06-02
+status: implementable
+see-also:
+  - https://github.com/operator-framework/operator-sdk/pull/2860
+---
+
+# README describing new command workflows for OLM enabled operators
+
+## Release Signoff Checklist
+
+- \[x\] Enhancement is `implementable`
+- \[x\] Design details are appropriately documented from clear requirements
+- \[x\] Test plan is defined
+- \[ \] Graduation criteria for dev preview, tech preview, GA
+- \[ \] User-facing documentation is created.
+
+## Open Questions:
+
+* _(Old project layouts)_ Should a user be able to generate both a bundle and
+package manifests format in the same directory (for migration purposes), or should
+they be mutually exclusive, ex. `generate bundle` will return an error if a
+`generate packagemanifests` has been run in the same output directory, and visa versa?
+
+## Summary
+
+`operator-sdk generate csv` is the current entrypoint into Operator packaging
+for developers of Operator SDK projects. This command attempts to handle both
+[bundle][bundle-format] and [package manifests][package-manifests-format] format
+generation scenarios with a plethora of flags. The number of flags and confusing
+command name (`generate csv` manages higher-level formats) creates non-ideal UX.
+This proposal details new workflows for bundle and package manifests formats
+for old and [new][ep-kb-integration] `operator-sdk`-build Operator projects.
+
+To clearly define what old and new layouts mean:
+* Old project layouts: scaffolded with `operator-sdk new`
+* New project layouts: scaffolded with `operator-sdk init`
+
+We plan to improve the developer experience for operator authors using Operator SDK by providing:
+1. One `operator-sdk` subcommand _per format_ for generating manifests and metadata.
+1. Interactive-by-default commands to get UI information for packaged manifests and metadata.
+
+[bundle-format]:https://github.com/operator-framework/operator-registry/blob/master/docs/design/operator-bundle.md#bundle-manifest-format
+[package-manifests-format]:https://github.com/operator-framework/operator-registry/tree/v1.5.3/#manifest-format
+[ep-kb-integration]:https://github.com/operator-framework/operator-sdk/blob/master/proposals/kubebuilder-integration.md
+
```

Comment:
Could we centralise all links at the bottom?

Reply:
Agreed since they're just references put them at the bottom. And I didn't know this was a feature. I will do this in my next proposal since I hate long lines everywhere :)